GuiltMotivator/app/src/main/java/nielsen/guiltmotivator/ContactAdapter.java

```diff
@@ -2,50 +2,130 @@
+            holder = new ContactHolder();
+            convertView.setTag(holder);
         }
+
+        holder.contact = getItem(position);
+
         // I didn't figure out how to implement ButterKnife here.
         // Because all elements here is a property of the holder class
+
         holder.name = (TextView) convertView.findViewById(R.id.itemName);
-        holder.method = (TextView)convertView.findViewById(R.id.itemMethod);
-        holder.delete = (ImageButton)convertView.findViewById(R.id.delete);
-        // create onClickListener for edit Button (goes to Fragment Ingredient
+        holder.method = (TextView) convertView.findViewById(R.id.itemMethod);
+        holder.delete = (ImageButton) convertView.findViewById(R.id.delete);
 
+        holder.name.setText(holder.contact.getName());
+        holder.method.setText(holder.contact.getMethod());
 
+        // create onClickListener to edit the contact info
+        holder.name.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(getContext());
+                LayoutInflater inflater = LayoutInflater.from(getContext());
+                final View dialogView = inflater.inflate(R.layout.dialog_create_contact, null);
+                //set up the spinner
+                final Spinner methodSpinner = (Spinner) dialogView.findViewById(R.id.contactMethodSpinner);
+                ArrayAdapter<CharSequence> methodAdapter = ArrayAdapter.createFromResource(getContext(), R.array.contact_methods_array,
+                        android.R.layout.simple_spinner_item);
+                methodAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+                methodSpinner.setAdapter(methodAdapter);
+                //set up the alert dialog actions
+                alertDialogBuilder.setView(dialogView)
+                        .setTitle("Edit Contact!")
+                        .setPositiveButton("Okay", new DialogInterface.OnClickListener() {
+                            @Override
+                            public void onClick(DialogInterface dialogInterface, int i) {
+                                EditText nameEditText = (EditText) dialogView.findViewById(R.id.editTextContactName);
+                                EditText addressEditText = (EditText) dialogView.findViewById(R.id.editTextContactAddress);
+
+                                nameEditText.setHint(holder.contact.getName());
+                                addressEditText.setHint(holder.contact.getAddress());
+
+                                String name = nameEditText.getText().toString();
+                                String address = addressEditText.getText().toString();
+                                String method = methodSpinner.getItemAtPosition(methodSpinner.getSelectedItemPosition()).toString();
+
```

Comment:
It's a long line but I don't think splitting it up would make it more clear in this case. Getting things from a spinner is just ugly.